### ISSUE TITLE
is it a good way to just calculate the number of leaves as the feature importance?(Python version)

### ISSUE THREAD

**Issue by PengFoo** [dmlc/xgboost#1329]
in xgboost.core (Python version), function get_fscore(self, fmap='') just calculate the number of leaves as the feature importance and return, even without a normalization.

is it a good way to measure the feature importance?  could it represent the real importance of the feature?

**Comment by bobye** [comment]
I think it is not. The feature importance is a way to measure how "non-linearly" each feature plays in constructing trees. The more they played the more likely they are overfitting. Therefore, you should also look at the gains and averaged gains of each feature, in order to gain a sense of importance. (Check this out https://github.com/Far0n/xgbfi )

Somewhat I feel the name of importance is a bit misleading. Anyway, you should take whatever proven to be useful in your validation experiment.

**Comment by PengFoo** [comment]
@bobye  
i agree that for the word "importance", the information gain is what measures how important the feature is, the more gain a feature brings, the more important the feature is.

so, it seems that the f_score function and the importance it returns actually does not mean a lot, am i right?

**Comment by jokari69** [comment]
I agree that the current python implementation only returns weight (while the R package has the gain)
It is possible to create it yourself by looking at bst= get_dump(with_stats=True) and adding up all the gain for each feature.
I can make a pull request to add this to python package it has the same functionality as the R package if this would be helpful.
